app/celery/process_ses_receipts_tasks.py

```diff
@@ -1,19 +1,80 @@
-from flask import current_app
+from datetime import datetime, timedelta
+
+import iso8601
+from flask import current_app, json
 from notifications_utils.statsd_decorators import statsd
+from sqlalchemy.orm.exc import NoResultFound
 
-from app import notify_celery
+from app import notify_celery, statsd_client
 from app.config import QueueNames
+from app.clients.email.aws_ses import get_aws_responses
+from app.dao import notifications_dao
+from app.models import NOTIFICATION_SENDING, NOTIFICATION_PENDING
 
-from app.notifications.notifications_ses_callback import process_ses_response
+from app.notifications.notifications_ses_callback import (
+    determine_notification_bounce_type,
+    handle_complaint,
+    _check_and_queue_complaint_callback_task,
+    _check_and_queue_callback_task,
+)
 
 
 @notify_celery.task(bind=True, name="process-ses-result", max_retries=5, default_retry_delay=300)
 @statsd(namespace="tasks")
 def process_ses_results(self, response):
     try:
-        errors = process_ses_response(response)
-        if errors:
-            current_app.logger.error(errors)
-    except Exception:
-        current_app.logger.exception('Error processing SES results')
+        ses_message = json.loads(response['Message'])
+        notification_type = ses_message['notificationType']
+
+        if notification_type == 'Bounce':
+            notification_type = determine_notification_bounce_type(notification_type, ses_message)
+        elif notification_type == 'Complaint':
+            _check_and_queue_complaint_callback_task(*handle_complaint(ses_message))
+            return True
+
+        aws_response_dict = get_aws_responses(notification_type)
+
+        notification_status = aws_response_dict['notification_status']
+        reference = ses_message['mail']['messageId']
+
+        try:
+            notification = notifications_dao.dao_get_notification_by_reference(reference)
+        except NoResultFound:
+            message_time = iso8601.parse_date(ses_message['mail']['timestamp']).replace(tzinfo=None)
```

Comment:
Is it guaranteed that the timestamp is in UTC?

Reply:
All the examples I've seen have a zero offset, so I think so. SES documentation doesn't mention it, but I think it might be implied.